helper/scaleutils/nomad.go

```diff
@@ -250,3 +268,38 @@ func (si *ScaleIn) monitorNodeDrain(ctx context.Context, nodeID string, index ui
 	}
 	return ctx.Err()
 }
+
+// identifyAutoscalerNodeID identifies the NodeID which the autoscaler is
+// running on.
+//
+// TODO(jrasell) this should be removed once the cluster targets and core
+//  autoscaler components are updated to handle reconciliation.
+func identifyAutoscalerNodeID(client *api.Client) (string, error) {
+
+	envVar := os.Getenv("NOMAD_ALLOC_ID")
+	if envVar == "" {
+		return "", nil
+	}
+
+	allocInfo, _, err := client.Allocations().Info(envVar, nil)
+	if err != nil {
+		return "", fmt.Errorf("failed to call Nomad allocations info: %v", err)
```

Comment:
my question is whether this should be best-effort, in which case `identifyAutoscalerNodeID` should never return an error and block a ScaleUtils from being instantiated... or whether this error (which suggests bad config) should be permitted.

i'm leaning towards best-effort, in which case we can just log any errors.

Reply:
Yeah that makes sense to me. I'll get that changed.